service/tablet_allocator.cc

```diff
@@ -990,26 +987,26 @@ class load_balancer {
 
             bool check_rack_load = false;
             bool has_replica_on_target = false;
-            std::unordered_map<sstring, int> rack_load; // Will be built if check_rack_load
+            std::unordered_map<const locator::rack*, int> rack_load; // Will be built if check_rack_load
 
             if (nodes_to_drain.empty()) {
-                check_rack_load = target_node.dc_rack().rack != topo.get_node(src.host).dc_rack().rack;
+                check_rack_load = target_node.location().rack() != topo.get_node(src.host).location().rack();
                 for (auto&& r: tmap.get_tablet_info(source_tablet.tablet).replicas) {
                     if (r.host == target) {
                         has_replica_on_target = true;
                         break;
                     }
                     if (check_rack_load) {
                         const locator::node& node = topo.get_node(r.host);
-                        if (node.dc_rack().dc == dc) {
-                            rack_load[node.dc_rack().rack] += 1;
+                        if (node.location().dc() == dc) {
+                            rack_load[node.location().rack()] += 1;
```

Comment:
This asks for short-circuiting location().rack() into rack() and same for dc().

Reply:
can do

Reply:
done in latest version